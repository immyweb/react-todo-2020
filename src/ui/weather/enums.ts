export enum Locations {
  WICK = 'WICK',
  ABERDEEN = 'ABERDEEN',
  EDINBURGH = 'EDINBURGH',
  BELFAST = 'BELFAST',
  MANCHESTER = 'MANCHESTER',
  CARDIFF = 'CARDIFF',
  PLYMOUTH = 'PLYMOUTH',
  LONDON = 'LONDON',
}

export enum Days {
  MON = 'MON',
  TUE = 'TUE',
  WED = 'WED',
  THU = 'THU',
  FRI = 'FRI',
  SAT = 'SAT',
  SUN = 'SUN',
}

export enum Direction {
  LEFT = 'LEFT',
  RIGHT = 'RIGHT',
  TOP = 'TOP',
  BOTTOM = 'BOTTOM',
}

export enum WeatherIconUrls {
  CLEAR_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/clear-night@2x1.png',
  SUNNY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png',
  PARTLY_CLOUDY_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png',
  PARTLY_CLOUDY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png',
  MIST = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/foggy@2x1.png',
  FOG = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/foggy@2x1.png',
  CLOUDY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Cloudy.png',
  OVERCAST = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Cloudy.png',
  LIGHT_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  LIGHT_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  DRIZZLE = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  LIGHT_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  HEAVY_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  HEAVY_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  HEAVY_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  SLEET_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  SLEET_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  SLEET = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HAIL_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Hail.png',
  HAIL_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Hail.png',
  HAIL = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Hail.png',
  LIGHT_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  LIGHT_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  LIGHT_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HEAVY_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HEAVY_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HEAVY_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  THUNDER_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Storm.png',
  THUNDER_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Storm.png',
  THUNDER = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Storm.png',
}

export enum WeatherAtmosphereUrls {
  CLEAR_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_clearnight@2x.jpg',
  SUNNY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_sunny@2x.jpg',
  PARTLY_CLOUDY_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_partlycloudynight@2x.jpg',
  PARTLY_CLOUDY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_partlycloudynight@2x.jpg',
  MIST = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_foggy@2x.jpg',
  FOG = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_foggy@2x.jpg',
  CLOUDY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_cloudy@2x.jpg',
  OVERCAST = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_cloudy@2x.jpg',
  LIGHT_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  LIGHT_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  DRIZZLE = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  LIGHT_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HEAVY_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HEAVY_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HEAVY_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  SLEET_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  SLEET_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  SLEET = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HAIL_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HAIL_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HAIL = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  LIGHT_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  LIGHT_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  LIGHT_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HEAVY_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HEAVY_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HEAVY_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  THUNDER_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_stormy@2x.jpg',
  THUNDER_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_stormy@2x.jpg',
  THUNDER = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_stormy@2x.jpg',
}
