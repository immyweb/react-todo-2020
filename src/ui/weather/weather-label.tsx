import * as React from 'react';

import { Direction } from './enums';
import { IPosition } from '../../types/position';
import { colours } from '../../styles/colours';
import {
  WeatherLabelContainer,
  WeatherNameContainer,
  WeatherArrow,
  WeatherArrowBottom,
  WeatherArrowTop,
  WeatherName,
  WeatherTemperature,
  WeatherTempHolder,
  WeatherIcon,
  WeatherIconBottom,
  WeatherIconTop,
} from './styles';

export interface IProps {
  dayMax: string;
  nightMin: string;
  direction: Direction;
  arrowDirection: Direction;
  icon: string;
  mobilePosition: IPosition;
  desktopPosition: IPosition;
}

export const WeatherLabel: React.FC<IProps> = ({
  dayMax,
  nightMin,
  direction,
  arrowDirection,
  icon,
  mobilePosition,
  desktopPosition,
  children,
}) => (
  <WeatherLabelContainer
    className="WeatherLabelContainer"
    direction={direction}
    mTop={mobilePosition.top}
    mLeft={mobilePosition.left}
    dTop={desktopPosition.top}
    dLeft={desktopPosition.left}
  >
    <WeatherNameContainer direction={direction}>
      {arrowDirection === Direction.LEFT && (
        <div style={{ display: 'flex' }}>
          <WeatherIcon className="WeatherLabelIcon" iconUrl={icon} />
          <WeatherArrow className="WeatherLabelArrow" backgroundColour={colours.$offBlack} direction={arrowDirection} />
        </div>
      )}
      {arrowDirection === Direction.RIGHT && (
        <div style={{ display: 'flex' }}>
          <WeatherArrow className="WeatherLabelArrow" backgroundColour={colours.$offBlack} direction={arrowDirection} />
          <WeatherIcon className="WeatherLabelIcon" iconUrl={icon} />
        </div>
      )}
      {arrowDirection === Direction.BOTTOM && (
        <div>
          <WeatherArrowBottom className="WeatherLabelArrowBottom" backgroundColour={colours.$offBlack} />
          <WeatherIconBottom className="WeatherLabelIcon" iconUrl={icon} />
        </div>
      )}
      {arrowDirection === Direction.TOP && (
        <div>
          <WeatherArrowTop className="WeatherLabelArrowTop" backgroundColour={colours.$offBlack} />
          <WeatherIconTop className="WeatherLabelIcon" iconUrl={icon} />
        </div>
      )}
      <WeatherName className="WeatherLabelName" labelColour={colours.$white} labelBackgroundColour={colours.$offBlack}>
        {children}
      </WeatherName>
    </WeatherNameContainer>
    <WeatherTempHolder>
      <WeatherTemperature
        className="WeatherLabelTemperature"
        temperatureColour={colours.$white}
        temperatureBackgroundColour={colours.$mainSecondary}
      >
        {`${dayMax}°`}
      </WeatherTemperature>
      <WeatherTemperature
        className="WeatherLabelTemperature"
        temperatureColour={colours.$white}
        temperatureBackgroundColour={colours.$newsSecondary}
      >
        {`${nightMin}°`}
      </WeatherTemperature>
    </WeatherTempHolder>
  </WeatherLabelContainer>
);
