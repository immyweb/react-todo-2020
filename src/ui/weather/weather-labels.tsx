import * as React from 'react';

import { WeatherLabel } from './weather-label';

import { WeatherIconUrls, Direction } from './enums';
import { ILocation } from './types';

export interface IProps {
  locations: ILocation[];
  selectedIndex: number;
  weekForecast: any;
}

export const WeatherLabels: React.FC<IProps> = ({ locations, selectedIndex, weekForecast }) => {
  const labels = locations.map((location, i) => {
    const { id, label, direction, arrowDirection, mobilePosition, desktopPosition } = location;

    return (
      <WeatherLabel
        key={i}
        dayMax={weekForecast[selectedIndex][id].day.DayMaximumTemperature}
        nightMin={weekForecast[selectedIndex][id].night.NightMinimumTemperature}
        direction={Direction[direction]}
        arrowDirection={Direction[arrowDirection]}
        icon={WeatherIconUrls[weekForecast[selectedIndex][id].day.WeatherType]}
        mobilePosition={mobilePosition}
        desktopPosition={desktopPosition}
      >
        {label}
      </WeatherLabel>
    );
  });
  return <React.Fragment>{labels}</React.Fragment>;
};
