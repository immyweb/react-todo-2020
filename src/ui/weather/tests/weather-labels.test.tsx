import * as React from 'react';
import { shallow } from 'enzyme';

import { WeatherLabels } from '../weather-labels';
import { WeatherLabel } from '../weather-label';

import { Direction, Locations } from '../enums';

const locations = [
  {
    id: 'WICK',
    label: Locations.WICK,
    direction: Direction.LEFT,
    position: {
      top: '5%',
      left: '15%',
    },
  },
  {
    id: 'LOND',
    label: Locations.LONDON,
    direction: Direction.RIGHT,
    position: {
      top: '15%',
      left: '5%',
    },
  },
];

const weekForecast = [
  {
    WICK: {
      date: 20190131,
      day: {
        WeatherType: 'SLEET',
        DayMaximumTemperature: '2',
        FeelsLikeDayMaximumTemperature: '-1',
        WindDirection: 'WSW',
        WindGustNoon: '9',
        ScreenRelativeHumidityNoon: '94',
        PrecipitationProbabilityDay: '78',
        WindSpeed: '9',
        Visibility: 'VG',
        MaxUVIndex: '1',
      },
      night: {
        WeatherType: 'CLOUDY',
        NightMinimumTemperature: '-1',
        FeelsLikeNightMinimumTemperature: '-4',
        WindDirection: 'W',
        WindGustMidnight: '16',
        ScreenRelativeHumidityMidnight: '96',
        PrecipitationProbabilityNight: '49',
        WindSpeed: '9',
        Visibility: 'VG',
      },
    },
    LOND: {
      date: 20190131,
      day: {
        WeatherType: 'PARTLY_CLOUDY_DAY',
        DayMaximumTemperature: '5',
        FeelsLikeDayMaximumTemperature: '1',
        WindDirection: 'E',
        WindGustNoon: '13',
        ScreenRelativeHumidityNoon: '75',
        PrecipitationProbabilityDay: '4',
        WindSpeed: '11',
        Visibility: 'GO',
        MaxUVIndex: '1',
      },
      night: {
        WeatherType: 'LIGHT_SNOW',
        NightMinimumTemperature: '1',
        FeelsLikeNightMinimumTemperature: '-4',
        WindDirection: 'E',
        WindGustMidnight: '27',
        ScreenRelativeHumidityMidnight: '88',
        PrecipitationProbabilityNight: '84',
        WindSpeed: '13',
        Visibility: 'MO',
      },
    },
  },
];

function shallowRender(props?: any) {
  return shallow(
    <WeatherLabels
      locations={locations}
      selectedIndex={0}
      weekForecast={weekForecast}
      {...props}
    />,
  );
}

describe('<WeatherLabels />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct number of labels', () => {
    const wrapper = shallowRender();
    const labels = wrapper.find(WeatherLabel);
    expect(labels.length).toEqual(2);
  });
});
