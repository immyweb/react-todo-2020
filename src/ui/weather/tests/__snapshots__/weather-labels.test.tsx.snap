// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherLabels /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    locations={
      Array [
        Object {
          "direction": "LEFT",
          "id": "WICK",
          "label": "WICK",
          "position": Object {
            "left": "15%",
            "top": "5%",
          },
        },
        Object {
          "direction": "RIGHT",
          "id": "LOND",
          "label": "LONDON",
          "position": Object {
            "left": "5%",
            "top": "15%",
          },
        },
      ]
    }
    selectedIndex={0}
    weekForecast={
      Array [
        Object {
          "LOND": Object {
            "date": 20190131,
            "day": Object {
              "DayMaximumTemperature": "5",
              "FeelsLikeDayMaximumTemperature": "1",
              "MaxUVIndex": "1",
              "PrecipitationProbabilityDay": "4",
              "ScreenRelativeHumidityNoon": "75",
              "Visibility": "GO",
              "WeatherType": "PARTLY_CLOUDY_DAY",
              "WindDirection": "E",
              "WindGustNoon": "13",
              "WindSpeed": "11",
            },
            "night": Object {
              "FeelsLikeNightMinimumTemperature": "-4",
              "NightMinimumTemperature": "1",
              "PrecipitationProbabilityNight": "84",
              "ScreenRelativeHumidityMidnight": "88",
              "Visibility": "MO",
              "WeatherType": "LIGHT_SNOW",
              "WindDirection": "E",
              "WindGustMidnight": "27",
              "WindSpeed": "13",
            },
          },
          "WICK": Object {
            "date": 20190131,
            "day": Object {
              "DayMaximumTemperature": "2",
              "FeelsLikeDayMaximumTemperature": "-1",
              "MaxUVIndex": "1",
              "PrecipitationProbabilityDay": "78",
              "ScreenRelativeHumidityNoon": "94",
              "Visibility": "VG",
              "WeatherType": "SLEET",
              "WindDirection": "WSW",
              "WindGustNoon": "9",
              "WindSpeed": "9",
            },
            "night": Object {
              "FeelsLikeNightMinimumTemperature": "-4",
              "NightMinimumTemperature": "-1",
              "PrecipitationProbabilityNight": "49",
              "ScreenRelativeHumidityMidnight": "96",
              "Visibility": "VG",
              "WeatherType": "CLOUDY",
              "WindDirection": "W",
              "WindGustMidnight": "16",
              "WindSpeed": "9",
            },
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Unknown
          dayMax="2"
          direction="LEFT"
          icon="https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png"
          nightMin="-1"
        >
          WICK
        </Unknown>,
        <Unknown
          dayMax="5"
          direction="RIGHT"
          icon="https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png"
          nightMin="1"
        >
          LONDON
        </Unknown>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "arrowDirection": undefined,
          "children": "WICK",
          "dayMax": "2",
          "desktopPosition": undefined,
          "direction": "LEFT",
          "icon": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png",
          "mobilePosition": undefined,
          "nightMin": "-1",
        },
        "ref": null,
        "rendered": "WICK",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "arrowDirection": undefined,
          "children": "LONDON",
          "dayMax": "5",
          "desktopPosition": undefined,
          "direction": "RIGHT",
          "icon": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png",
          "mobilePosition": undefined,
          "nightMin": "1",
        },
        "ref": null,
        "rendered": "LONDON",
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Unknown
            dayMax="2"
            direction="LEFT"
            icon="https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png"
            nightMin="-1"
          >
            WICK
          </Unknown>,
          <Unknown
            dayMax="5"
            direction="RIGHT"
            icon="https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png"
            nightMin="1"
          >
            LONDON
          </Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "arrowDirection": undefined,
            "children": "WICK",
            "dayMax": "2",
            "desktopPosition": undefined,
            "direction": "LEFT",
            "icon": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png",
            "mobilePosition": undefined,
            "nightMin": "-1",
          },
          "ref": null,
          "rendered": "WICK",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "arrowDirection": undefined,
            "children": "LONDON",
            "dayMax": "5",
            "desktopPosition": undefined,
            "direction": "RIGHT",
            "icon": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png",
            "mobilePosition": undefined,
            "nightMin": "1",
          },
          "ref": null,
          "rendered": "LONDON",
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
